shared/chain_support.go

```diff
@@ -31,11 +31,27 @@ type ChainSupport interface {
 type CCIPChainSupport struct {
 	lggr            logger.Logger
 	homeChain       reader.HomeChain
-	oracleIDToP2pID map[commontypes.OracleID]libocrtypes.PeerID
+	oracleIDToP2PID map[commontypes.OracleID]libocrtypes.PeerID
 	nodeID          commontypes.OracleID
 	destChain       cciptypes.ChainSelector
 }
 
+func NewCCIPChainSupport(
+	lggr logger.Logger,
+	homeChain reader.HomeChain,
+	oracleIDToP2PID map[commontypes.OracleID]libocrtypes.PeerID,
+	nodeID commontypes.OracleID,
+	destChain cciptypes.ChainSelector,
+) CCIPChainSupport {
+	return CCIPChainSupport{
+		lggr:            lggr,
+		homeChain:       homeChain,
+		oracleIDToP2PID: oracleIDToP2PID,
+		nodeID:          nodeID,
+		destChain:       destChain,
+	}
+}
+
```

Comment:
Maybe this thing could be passed into the plugin instead of the home chain? And/or expanded to support all the subconsensus stuff.

Reply:
Might worth checking the feasability of this 👍